src/Mod/Part/App/PropertyTopoShape.cpp

```diff
@@ -271,11 +271,14 @@ void PropertyPartShape::Save (Base::Writer &writer) const
     }
     std::string version;
     // If exporting, do not export mapped element name, but still make a mark
-    if(owner) {
-        if(!owner->isExporting())
-            version = _Ver.size()?_Ver:owner->getElementMapVersion(this);
-    }else
-        version = _Ver.size()?_Ver:_Shape.getElementMapVersion();
+    auto const version_valid = _Ver.size() && (_Ver != "?");
+    if (owner) {
+        if (!owner->isExporting()) {
+            version = version_valid ? _Ver : owner->getElementMapVersion(this);
+        }
+    } else {
+        version = version_valid ? _Ver : _Shape.getElementMapVersion();
+    }
     writer.Stream() << " ElementMap=\"" << version << '"';

```

Comment:
For reference, this code block can be located here in @realthunder's version: https://github.com/realthunder/FreeCAD/blob/a9810d509a6f112b5ac03d4d4831b67e6bffd5b7/src/Mod/Part/App/PropertyTopoShape.cpp#L319-L325 and it looks exactly like the original code.